### ISSUE TITLE
some sections of page are hidden

### ISSUE THREAD

**Issue by ranjitgalsar** [locomotivemtl/locomotive-scroll#209]
Locomotive smooth scroll hides some sections of the page at the bottom, or sometimes the whole page is crashed. it happens in my all websites.

**Comment by TonyCastillo** [comment]
any solutions for this issue?? The same happens to me.  But I see when the windows, i mean, when the explorer is resize is working fine.

**Comment by umberto02** [comment]
Hi, I also have the same problem.
Is there a solution?

Thanks

**Comment by remi-lebigre** [comment]
From what I understand, locomotivescroll calculates your page height on initialization. Some content may have not finished to load and change page height afterwards (for example image loading).
When implementing locomotivescroll, I ensure all images have been downloaded and I set a small timeout before initializing the library.
I am not sure this is the right way to do though, I would gladly hear the maintainers' point of view.

**Comment by ranjitgalsar** [comment]
use setTimeout with DOMContentLoaded event is working sometimes but it's not an ideal way.

**Comment by ranjitgalsar** [comment]
> any solutions for this issue?? The same happens to me. But I see when the windows, i mean, when the explorer is resize is working fine.

use setTimeout with DOMContentLoaded event is working sometimes but it's not an ideal way.

**Comment by umberto02** [comment]
Hi, can you post an example.

Thanks for your help

**Comment by remi-wakeo** [comment]
Here is my current implementation:

```
initScrollOnImageLoad = _ => {
    Promise.all([...this.container.querySelectorAll('img')]
      .filter(img => !img.complete)
      .map(img => new Promise(resolve => {
        img.onload = img.onerror = resolve
      }))).then(this.initScroll)
  }
 initScroll = _ => {
    setTimeout(_ => {
      this.locomotive = new LocomotiveScroll({
        el: this.container.querySelector('[data-scroll-container]'),
        smooth: true,
      })
    }, 200)
  }
```
I have not found a better way than adding an arbitrary 200ms delay.

**Comment by TonyCastillo** [comment]
Yes, it's like Mario says it. locomotivescroll calculates your page height on initialization. So I implent this on my project and is working fine.

`window.addEventListener("load", function(event) {
      let lazyScroll = new LocomotiveScroll({
        el: document.getElementById("main-content-scroll"),
        smooth: true
      });`

**Comment by umberto02** [comment]
I finally figured out the error between the height and image upload. But I still have the same problem, as using GSAP Scrolltrigger I can't integrate your scripts.

I'll give you an example of my structure:

const locoScroll = new LocomotiveScroll({
  el: document.querySelector("#js-scroll"),
  smooth: true,
  //inertia: .6,
  lerp: .07,
    
});

// each time Locomotive Scroll updates, tell ScrollTrigger to update too (sync positioning)
locoScroll.on("scroll", ScrollTrigger.update);

// tell ScrollTrigger to use these proxy methods for the ".smooth-scroll" element since Locomotive Scroll is hijacking things
ScrollTrigger.scrollerProxy("#js-scroll", {
  scrollTop(value) {
    return arguments.length ? locoScroll.scrollTo(value, 0, 0) : locoScroll.scroll.instance.scroll.y;
  }, // we don't have to define a scrollLeft because we're only scrolling vertically.
  getBoundingClientRect() {
    return {top: 0, left: 0, width: window.innerWidth, height: window.innerHeight};
  },
  // LocomotiveScroll handles things completely differently on mobile devices - it doesn't even transform the container at all! So to get the correct behavior and avoid jitters, we should pin things with position: fixed on mobile. We sense it by checking to see if there's a transform applied to the container (the LocomotiveScroll-controlled element).
  pinType: document.querySelector("#js-scroll").style.transform ? "transform" : "fixed"
});
	 

// each time the window updates, we should refresh ScrollTrigger and then update LocomotiveScroll. 
ScrollTrigger.addEventListener("refresh", () => locoScroll.update());

// after everything is set up, refresh() ScrollTrigger and update LocomotiveScroll because padding may have been added for pinning, etc.
ScrollTrigger.refresh();


Can you help me?

Thanks
